### DISCUSSION TITLE
encryption/decryption with multiple keys (multiple recipients)

### DISCUSSION THREAD

**Discussion by valsha** [JTD420/laravel-pgp#4] [Ideas]
Can U add the ability to use multiple recipients for messages (PGP)?

Found this as an example:

https://stackoverflow.com/questions/597188/encryption-decryption-with-multiple-keys

Thank you.

**Comment by JTD420** [comment]
> Can U add the ability to use multiple recipients for messages (PGP)?
> 
> Found this as an example:
> 
> https://stackoverflow.com/questions/597188/encryption-decryption-with-multiple-keys
> 
> Thank you.

Hey 👋

Thanks for another good question and for your continued interest in my Laravel package!

Regarding your question about multiple recipients for PGP messages, I'd like to know more about how you plan to use the package for messages?

If you intend to use the provided chat interface, then it will be possible to send messages to multiple recipients. However, please note that the package is not currently using multi-key encryption, but rather looping through the recipient_ids and encrypting the message to each recipient separately. This is because I was unable to find a way to do multi-key encryption and wanted to get a stable working version out.

If you're interested in how I'm handling it in the MessageController class, I'll be happy to share the code with you once it's released. I'll comment here again with a link to the relevant line in the code once it's pushed to the repository.

Thanks again for your support, and feel free to let me know if you have any other questions or concerns.

-JTD

**Comment by JTD420** [comment]
_**ToDo:** 
Look into how multiple keys are handled with openpgp-php_ RE:
https://github.com/singpolyma/openpgp-php/issues/41#issuecomment-292729600

**Example** : https://github.com/singpolyma/openpgp-php/blob/master/examples/encryptDecrypt.php

**Comment by valsha** [comment]
Hi, yes, we would like to use this feature to send notifications to all our users, such as newsletters in our chat, etc.
